### PR TITLE
Include country in add patient mutation; add Canadian state/zip selection

### LINKED ISSUE #3201: SPIKE: Add State/ Province and Postal Code capture to International Addresses
One of the border states using SimpleReport shared some issues with the way we are capturing international addresses. They would like to see State/ Province and Postal Code reflected. This is a change we can make, and we should coordinate with RS to make sure that the proper translation is implemented when these records are sent. 

Please see the email below --- 

One of our facilities near the Canadian border occasionally does testing for Canadian residents visiting Vermont. In the HL7 messages for results entered into SimpleReport for these patients, we are seeing default values of Patient Address State (PID-11.4) = “NA”, Patient Address ZIP or Postal Code (PID-11.5) = “00000”, and Patient Country (PID-11.6) = “USA”.

 

I tried creating a Canadian patient in the SimpleReport demo site, and when I select Country = Canada, the state/province and ZIP code fields disappear and there is no option for a person entering data to select a state/province or enter a ZIP code.

 

I have two requests related to this:

When Patient Country = Canada in SimpleReport, the patient country should be set to “CAN” in PID-11.6.
Is it possible to collect state/province and postal code information for Canadian residents?
If so, please send this in those fields, ensuring correct province abbreviations per HL7 guidelines.
If that information is not able to be collected at this time, please send PID-11.4 and PID-11.5 blank instead of using the default values of “NA” and “00000”.
 

Here are a few example accession numbers for records from today with these issues:

ba3ca9e2-aa2d-4bd6-b2b0-81c1cd2caeac

4c595ca2-5825-4b3c-961d-d46000fe2f2c

d27bf370-80af-4032-a23a-2885492b63bc

4e62f2ee-be86-4ab1-bcc4-bd3d60a73e3d

 

Thank you!

### PR DESCRIPTION
**Pull request by emmastephenson**
## Related Issue or Background Info

- fixes #3201 

## Changes Proposed

- add a list of Canadian provinces as the dropdown when CAN is selected as the patient country
- fix add patient frontend to correctly store patient country

## Screenshots / Demos

Screen Shot 2022-01-24 at 3 22 09 PM

Screen Shot 2022-01-24 at 3 22 13 PM

### Infrastructure
- [ ] **Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!**

### Design
- [ ] Any UI/UX changes have a designer as a reviewer, and changes have been approved
- [x] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke-tested by both the engineering and design teams

### Content
- [x] Any content changes (including new error messages) have been approved by content team

### Support
- [x] Any changes that might generate new support requests have been flagged to the support team
- [x] Any changes to support infrastructure have been demo'd to support team

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [x] Database changes are submitted as a separate PR
  - [x] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [x] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [x] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [x] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [x] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [x] Any dependencies introduced have been vetted and discussed

## Cloud
- [x] DevOps team has been notified if PR requires ops support
- [x] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

**Review by nickclyde — APPROVED** [review]
Looks great! Thanks for this fix Emma!

File: backend/src/main/java/gov/cdc/usds/simplereport/api/Translators.java [hunk 1]
```diff
@@ -281,13 +281,16 @@ public static Boolean parseYesNo(String v) {
...
+      Set.of("AB", "BC", "MB", "NB", "NL", "NT", "NS", "NU", "ON", "PE", "QC", "SK", "YT");
+
   public static String parseState(String s) {
     String state = parseString(s);
     if (state == null) {
       return null;
     }
     state = state.toUpperCase();
-    if (STATE_CODES.contains(state)) {
+    if (STATE_CODES.contains(state) || CANADIAN_STATE_CODES.contains(state)) {
```
Comment (fzhao99) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/Translators.java:293]: nit: my brain did only hang for a half second on this, but for the sake of improving readability by a tiny bit/consistency with the new constant, would consider renaming STATE_CODES --> US_STATE_CODES

**Review by fzhao99 — APPROVED** [review]
Did not realize the useTranslation() hook existed as we're using it in the frontend patients component, but it looks super useful! Definitely filing away for future reference 🙂

File: backend/src/main/java/gov/cdc/usds/simplereport/api/Translators.java (hunk 1 above)
Comment (emmastephenson) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/Translators.java:293]: I would normally agree but I think we do use it in a couple other places, so for the sake of getting this fix in I'm going to keep it as-is. A good idea to change in the future though!

**Pull request merged**